cmd/root.go

```diff
@@ -3,66 +3,69 @@ package cmd
 import (
 	"context"
 
-	"github.com/google/go-github/github"
 	"github.com/hellofresh/github-cli/pkg/config"
-	"github.com/hellofresh/github-cli/pkg/formatter"
-	"github.com/pkg/errors"
-	log "github.com/sirupsen/logrus"
+	"github.com/hellofresh/github-cli/pkg/github"
+	"github.com/hellofresh/github-cli/pkg/log"
+	"github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
-	"golang.org/x/oauth2"
 )
 
-var (
-	cfgFile      string
-	globalConfig *config.Spec
-	githubClient *github.Client
-	token        string
-	verbose      bool
+type (
+	// RootOptions represents the ahoy global options
+	RootOptions struct {
+		configFile string
+		token      string
+		org        string
+		verbose    bool
+	}
+)
 
-	// RootCmd is our main command
-	RootCmd = &cobra.Command{
+// NewRootCmd creates the root command
+func NewRootCmd() *cobra.Command {
+	opts := RootOptions{}
+
+	cmd := cobra.Command{
 		Use:   "github-cli [--config] [--token]",
 		Short: "HF Github is a cli tool to manage your github repositories",
-		Long: `A simple CLI tool to help you manage your github repositories.
-		Complete documentation is available at http://github.com/hellofresh/github-cli`,
+		PersistentPreRun: func(ccmd *cobra.Command, args []string) {
+			if opts.verbose {
+				log.WithContext(context.Background()).SetLevel(logrus.DebugLevel)
+			}
+		},
 	}
-)
 
-func init() {
-	RootCmd.PersistentFlags().StringVarP(&cfgFile, "config", "c", "", "config file (default is $HOME/.github.toml)")
-	RootCmd.PersistentFlags().StringVarP(&token, "token", "t", "", "optional, github token for authentication (default in $HOME/.github.toml)")
-	RootCmd.PersistentFlags().BoolVarP(&verbose, "verbose", "v", false, "Make the operation more talkative")
+	cmd.PersistentFlags().StringVarP(&opts.configFile, "config", "c", "", "config file (default is $HOME/.github.toml)")
+	cmd.PersistentFlags().StringVarP(&opts.token, "token", "t", "", "optional, github token for authentication (default in $HOME/.github.toml)")
+	cmd.PersistentFlags().BoolVarP(&opts.verbose, "verbose", "v", false, "Make the operation more talkative")
+	cmd.PersistentFlags().StringVarP(&opts.org, "organization", "o", "", "Github's organization")
 
-	// Aggregates Root commands
-	RootCmd.AddCommand(NewRepoCmd())
-	RootCmd.AddCommand(NewHiringCmd())
-	RootCmd.AddCommand(NewVersionCmd())
-
-	log.SetFormatter(&formatter.CliFormatter{})
-}
-
-func setupConnection(cmd *cobra.Command, args []string) error {
-	var err error
+	ctx := log.NewContext(context.Background())
+	ctx, err := config.NewContext(ctx, opts.configFile)
+	if err != nil {
+		log.WithContext(ctx).WithError(err).Error("Could not load configuration file")
```

Comment:
should we return err here or it's fine to continue?

Reply:
It's fine to not have the config file. You should still be able to create a repository with the defaults